src/plugins/workspace/public/components/workspace_creator/workspace_permission_setting_panel.tsx

```diff
@@ -359,10 +359,36 @@ export const WorkspacePermissionSettingPanel = ({
       (permissionSettingItem) => permissionSettingItem.type === WorkspacePermissionItemType.Group
     ) ?? []
   );
+  const [userNonDeletableIndex, setUserNonDeletableIndex] = useState<number>(-1);
+  const [groupNonDeletableIndex, setGroupNonDeletableIndex] = useState<number>(-1);
```

Comment:
These two states are derived values from `userPermissionSettings`, `groupPermissionSettings`, I do not think it is a good practice to declare these two states. You can get these two value by using
```
const { userNonDeletableIndex, groupNonDeletableIndex } = useMemo(() => {}, [userPermissionSettings, groupPermissionSettings]);
```

Reply:
Thanks, I will update this.